dbms/src/Storages/DeltaMerge/SkippableBlockInputStream.h

```diff
@@ -25,10 +27,21 @@ namespace DM
 class SkippableBlockInputStream : public IBlockInputStream
 {
 public:
-    virtual ~SkippableBlockInputStream() = default;
+    ~SkippableBlockInputStream() override = default;
 
     /// Return false if it is the end of stream.
     virtual bool getSkippedRows(size_t & skip_rows) = 0;
+
+    /// Skip next block in the stream.
+    /// The number of rows of the next block shoule equal to skip_rows.
+    /// Return false if failed to skip or the end of stream.
+    virtual bool skipNextBlock(size_t skip_rows) = 0;
```

Comment:
`IBlockInputStream` should be a more suitable place for these two methods. And there is already a `read` method in `IBlockInputStream` which can be used to perform the role of `readWithFilter`.
`SkippableBlockInputStream` is mainly used to provide some info for the process of creating delta index. The "Skip" in `SkippableBlockInputStream` and "skip" in `skipNextBlock` is two different concepts which may cause some confusion for later maintainance.

Reply:
`IBlockInputStream` is the interface of all `BlockInputStream`, it may not suitable. And we can not combine `read` and `readWithFilter`.

Reply:
When does the `skipNextBlock` is called? And according to the comment, I believe `skip_rows` needs a better name.

Reply:
I will create two stream to read different columns, s1 and s2. When s1 read a block, and then executing filter condition, when all rows of the block are filtered-out, s2 will call `skipNextBlock`. Will change `bool skipNextBlock(size_t skip_rows)` to `size_t skipNextBlock()`, return the skipped rows.

Reply:
Add `dbms/src/Storages/DeltaMerge/LateMaterializationBlockInputStream.cpp`, you can check the usage of the two functions here.